inspirehep/modules/authors/static/js/authors/publications.js

```diff
@@ -0,0 +1,219 @@
+/*
+ * This file is part of INSPIRE.
+ * Copyright (C) 2016 CERN.
+ *
+ * INSPIRE is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * INSPIRE is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+* In applying this license, CERN does not
+* waive the privileges and immunities granted to it by virtue of its status
+* as an Intergovernmental Organization or submit itself to any jurisdiction.
+*/
+
+define(['publications'], function(Publications) {
+  var app = angular.module('publications', []);
+
+  app.controller('PublicationsController', ['$http', '$scope', 
+      function($http, $scope) {
+
+    $scope.publications = {
+      collaborations: [],
+      keywords: [],
+      publications: []
+    };
+
+    $scope.statistics = [[], 0];
+
+    var countPublications = function(data) {
+      var publications = data.publications;
+      var publicationsMap = {};
+      var publicationsDataToPlot = [];
+
+      var researchWorks = publications.length;
+
+      angular.forEach(publications, function(publication) {
+        if (publication.year in publicationsMap) {
+          publicationsMap[publication.year]++;
+        } else {
+          publicationsMap[publication.year] = 1;
+        }
+      });
+
+      angular.forEach(publicationsMap, function(count, year) {
+        publicationsDataToPlot.push({
+          year: parseInt(year),
+          publications: count
+        });
+      });      
+
+      return [publicationsDataToPlot, researchWorks];
+    };
+
+    var recid = $scope.record.control_number;
+    $http.get("/author/publications?recid=" + recid)
```

Comment:
@jmartinm this one is called every time when there is a reference form a directive.